ReSharper.FSharp/src/FSharp.Psi.Features/src/FSharpTypingAssist.fs

```diff
@@ -1493,6 +1495,47 @@ type FSharpTypingAssist(lifetime, solution, settingsStore, cachingLexerService,
 
         insertCharInBrackets context lexer atChars typedQuotationBrackets LeftBracketOnly.No
 
+    member x.HandleXmlDocTag(context: ITypingContext) =
+        let textControl = context.TextControl
+        let offset = textControl.Caret.Offset()
+        let mutable lexer = Unchecked.defaultof<_>
+        if not (x.GetCachingLexer(textControl, &lexer)) then false else
+
+        if offset < 4 then false else
+
+        // Check if new doc-comment block is being started
+        context.CallNext()
+        let file = x.CommitPsiOnlyAndProceedWithDirtyCaches(textControl, id)
+
+        let tokenNode = file.FindTokenAt(TreeOffset(offset)).As<DocComment>()
+        if isNull tokenNode then true else
+
+        let docCommentBlockNode = tokenNode.Parent.As<IDocCommentBlock>()
+
+        if isNull docCommentBlockNode ||
+           docCommentBlockNode.GetTextLength() < 4 ||
+           not docCommentBlockNode.IsSingleLine ||
+           tokenNode.CommentText.Trim() <> "<" then true else
+
+        let docCommentBlockOffset = docCommentBlockNode.GetTreeStartOffset().Offset
```

Comment:
Would getting `DocumentOffset` instead of `TreeOffset` simplify things below?

Reply:
No, because it also requires getting an offset (: